src/util/OAuth2Util.js

```diff
@@ -31,14 +31,20 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
    * @param params - {idp: 'xxx'} for social auth
    *                 {sessionToken: 'xxx'} for okta idp
    */
-  util.getTokens = function (settings, params) {
+  util.getTokens = function (settings, params, controller) {
 
     function success(result) {
       settings.callGlobalSuccess(Enums.SUCCESS, result);
     }
 
     function error(error) {
-      settings.callGlobalError(new Errors.OAuthError(error.message));
+      // OKTA-104330- Handle error case where user is not assigned to OIDC client
+      if (error.errorCode == 'access_denied') {
+        controller.model.trigger('error', controller.model, {'responseJSON': error});
```

Comment:
I agree that we're already bleeding internal state into OAuth2Util by exposing the settings object, but I'd prefer not to move even further down this route by passing in controller as well. Better to throw an error here, catch it in RouterUtil, and then call the controller stuff there.